src/main/java/io/debezium/connector/yugabytedb/YugabyteDBConnectorConfig.java

```diff
@@ -1433,6 +1534,165 @@ private static int validateLogicalDecodingMessageExcludeList(Configuration confi
+                    try {
+                        final Metronome retryMetronome = Metronome.parker(Duration.ofMillis(config.getLong(RETRY_DELAY_MS)), Clock.SYSTEM);
+                        retryMetronome.pause();
+                    } catch (InterruptedException ie) {
+                        LOGGER.warn("Connector retry sleep interrupted by exception: {}", ie);
+                        Thread.currentThread().interrupt();
+                    }
+                }
+            }
+        }
+
+        if (exception != null) {
+            throw exception;
+        }
+
+        return "";
+    }
+
+    public static String extractTableListFromPublication(Configuration config) {
+        String publicationName = config.getString(PUBLICATION_NAME);
+        String autoCreateMode = config.getString(PUBLICATION_AUTOCREATE_MODE);
+        Exception exception = null;
+
+        int retryCount = 0;
+        while (retryCount <= config.getInteger(MAX_CONNECTOR_RETRIES)) {
+            try (YugabyteDBConnection ybConnection = new YugabyteDBConnection(getJdbcConfig(config), YugabyteDBConnection.CONNECTION_GENERAL)) {
+                if (autoCreateMode.equalsIgnoreCase("disabled")) {
+                    if (!checkIfPublicationExists(publicationName, ybConnection)) {
+                        throw new DebeziumException(
+                         "Publication autocreation is disabled and the provided publication does not exist. Create the publication and restart the connector or change the publication.autocreate.mode");
+                    } else {
+                        return fetchTableIncludeList(ybConnection, publicationName);
+                    }             
+                } else if (autoCreateMode.equalsIgnoreCase("all tables")) {
+                    String query;
+                    if (!checkIfPublicationExists(publicationName, ybConnection)) {
+                        query = "CREATE PUBLICATION " + publicationName + " FOR ALL TABLES;";
+                        try (Connection connection = ybConnection.connection()) {
+                            ybConnection.execute(query);
+                        }
+                    } else {
+                        LOGGER.info("The publication {} already exists", publicationName);
+                    }
+                    return fetchTableIncludeList(ybConnection, publicationName);
+                } else if (autoCreateMode.equalsIgnoreCase("filtered")) {
+                    String tableList = config.getString(TABLE_INCLUDE_LIST);
+                    if (tableList == null || tableList.isEmpty()) {
+                        throw new DebeziumException("No tables included for Filtered Publication");
+                    }
+
+                    String createQuery = "CREATE PUBLICATION " + config.getString(PUBLICATION_NAME) + " FOR TABLE " + tableList + ";";
+                    String alterQuery = "ALTER PUBLICATION " + config.getString(PUBLICATION_NAME) + " SET TABLE " + tableList + ";";
```

Comment:
Was this "Alter publication" part also taken from the upstream connector?

Reply:
No, what I meant is I haven't borrowed this code from the Debezium PG connector. Instead I have implemented the behaviour as mentioned in the documentation [here](https://debezium.io/documentation/reference/1.9/connectors/postgresql.html#postgresql-publication-autocreate-mode)

Reply:
@Sumukh-Phalgaonkar We should check what the PG connector does for this. Its better to borrow their code/approach as that would be time tested.

Reply:
Yes, we are using the similar approach as PG connector. They also run `ALTER PUBLICATION`